### PR TITLE
RD-1964: move some TokenRequest logic to core so that user can use it

### PR DESCRIPTION
**Pull request by gianluca-token**

### CODE REVIEW

File: core/src/main/java/io/token/TokenClient.java [hunk 1]
```diff
@@ -580,6 +584,72 @@ public MemberRecoveryOperation getRecoveryAuthorizationBlocking(
...
+    }
+
+    /**
+     * Updates an existing token request.
+     *
+     * @param requestId token request ID
+     * @param options new token request options
+     * @return completable
+     */
+    public Completable updateTokenRequest(String requestId, TokenRequestOptions options) {
```
Comment (yuhanniu) [inline core/src/main/java/io/token/TokenClient.java]: Do TPPs need to update the token request?

**Comment by yuhanniu** [comment]
missing version change?

File: build.gradle [hunk 2]
```diff
@@ -32,5 +32,5 @@ allprojects {
     }
 
     group = 'io.token.sdk'
-    version = '2.0.0-Beta'
+    version = '2.0.1-Beta'
```
Comment (yuhanniu) [inline build.gradle]: `2.0.0-Beta2`?

File: tpp/src/main/java/io/token/tpp/tokenrequest/TokenRequestCallbackParameters.java [hunk 3]
```diff
@@ -30,6 +30,7 @@
 import io.token.tpp.exceptions.InvalidTokenRequestQuery;
 import io.token.tpp.util.Util;
 
+import java.security.KeyPair;
```
Comment (yuhanniu) [inline tpp/src/main/java/io/token/tpp/tokenrequest/TokenRequestCallbackParameters.java]: remove

File: core/src/main/java/io/token/rpc/UnauthenticatedClient.java [hunk 4]
```diff
@@ -27,10 +27,12 @@
 import static io.token.proto.common.security.SecurityProtos.Key.Level.LOW;
 import static io.token.proto.common.security.SecurityProtos.Key.Level.PRIVILEGED;
 import static io.token.proto.common.security.SecurityProtos.Key.Level.STANDARD;
+import static io.token.rpc.util.Converters.toCompletable;
 import static io.token.util.Util.generateNonce;
 import static io.token.util.Util.normalizeAlias;
 import static io.token.util.Util.toObservable;
 
+import io.reactivex.Completable;
```
Comment (yuhanniu) [inline core/src/main/java/io/token/rpc/UnauthenticatedClient.java]: remove unused imports

File: tpp/src/main/java/io/token/tpp/rpc/UnauthenticatedClient.java [hunk 5]
```diff
@@ -93,6 +93,21 @@ public Member apply(GetMemberResponse response) {
...
     }
 
+    /**
+     * Updates an existing token request.
+     *
+     * @param requestId token request ID
+     * @param options new token request options
+     * @return completable
+     */
+    public Completable updateTokenRequest(String requestId, TokenRequestOptions options) {
```
Comment (yuhanniu) [inline tpp/src/main/java/io/token/tpp/rpc/UnauthenticatedClient.java]: Can you remind me why do TPPs need to update the token requests?

Comment (gianluca-token) [reply tpp/src/main/java/io/token/tpp/rpc/UnauthenticatedClient.java]: ah I deleted it but forgot to commit. There's not a use case for it now; if there is, we can add it later.

File: build.gradle (hunk 2 above)
Comment (gianluca-token) [reply build.gradle]: changed it to 2.0.0-beta-2, a la https://mvnrepository.com/artifact/junit/junit

**Pull request merged**